lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/NodeStruct.cs

```diff
@@ -17,17 +17,22 @@
  * under the License.
  */
 
+using System;
 using System.Collections.Concurrent;
 using Org.Apache.REEF.Network.Group.Driver.Impl;
 
 namespace Org.Apache.REEF.Network.Group.Task.Impl
 {
+
     /// <summary>
     /// Stores all incoming messages sent by a particular Task.
+    /// Writable version
     /// </summary>
-    internal class NodeStruct
+    /// <typeparam name="T"> Generic type of message</typeparam>
+    [Obsolete("Need to remove Iwritable and use IstreamingCodec. Please see Jira REEF-295 ", false)]
+    internal sealed class NodeStruct<T>
     {
-        private readonly BlockingCollection<GroupCommunicationMessage> _messageQueue;
+        private readonly BlockingCollection<GroupCommunicationMessage<T>> _messageQueue;
```

Comment:
No this one is a GroupCommunicationMessage. So the semantics is that NetworkService layer sees GeneralGroupCommunicationMessage while Operator layer sees GroupCommunicationMessage.